### PR TITLE
ImageUtils.getDataURL: Added warning when saving image as jpg.

### PR DESCRIPTION
**Pull request by mrdoob**
**Description**

Log a warning when converting a image to jpeg due to its dimensions.
Seems like I forgot what issue/pr discussed this a few days ago. @Mugen87 do you remember?

### CODE REVIEW

**Comment by Mugen87** [comment]
> Seems like I forgot what issue/pr discussed this a few days ago. @Mugen87 do you remember?

No sorry. But the warning does make sense.

File: src/extras/ImageUtils.js [hunk 1]
```diff
@@ -47,6 +47,8 @@ const ImageUtils = {
 
 		if ( canvas.width > 2048 || canvas.height > 2048 ) {
 
+			console.warn( 'THREE.ImageUtils.getDataURL: Image converted to jpg for performance reasons', image );
```
Comment (Mugen87) [inline src/extras/ImageUtils.js:50]: Is it necessary to log the image object here? I remember there were performance issues in the past when doing this.

Comment (mrdoob) [reply src/extras/ImageUtils.js:50]: Hmm... Maybe `image.src` instead?

Comment (Mugen87) [reply src/extras/ImageUtils.js:50]: `image.src` won't work if image is of type `ImageBitmap`

Comment (Mugen87) [reply src/extras/ImageUtils.js:50]: I think we can start like so and remove the property if somebody complains. Unfortunately, I can't find the related discussion anymore so it's probably safe anyway.

**Review by Mugen87 — APPROVED** [review]

File: src/extras/ImageUtils.js (hunk 1 above)
Comment (mrdoob) [reply src/extras/ImageUtils.js:50]: Sounds good!

**Comment by gsimone** [comment]
Hey 👋 I just stumbled onto this, should we think about making this optimization skippable?

This lets users select the output format. Of course one can just patch this function away, but it's weird how opinionated the default is – I'm also missing the context here, so forgive me 🙏 

```js
getDataURL: function ( image, format ) {

    ...
    if (typeof format === "undefined") {

      if ( canvas.width > 2048 || canvas.height > 2048 ) {

        console.warn( 'THREE.ImageUtils.getDataURL: Image converted to jpg for performance reasons', image );

        return canvas.toDataURL( 'image/jpeg', 0.6 );

      } else {

        return canvas.toDataURL( 'image/png' );

      }
      
    } else {

      if (format === "png") {

         return canvas.toDataURL( 'image/png' );
         
      } else {
        
        return canvas.toDataURL( 'image/jpeg', 0.6 );
        
      }
  }
  ...

}
```

**Comment by mrdoob** [comment]
@gsimone do you mind detailing a bit what your use case is?

**Comment by gsimone** [comment]
We use Texture.toJson to serialize user-uploaded textures that might exceed 2048x2048, so a user uploaded a transparent png that was like 5k * 1k and it lost alpha information when it was serialized. We can of course work around this, but it's something that people might stumble into, might be worth at least documenting — although I'm still not sure why it's necessary

**Pull request merged**